SU2_CFD/src/numerics/NEMO/NEMO_sources.cpp

```diff
@@ -372,12 +372,12 @@ CNumerics::ResidualType<> CSource_NEMO::ComputeAxisymmetric(const CConfig *confi
                                                                     -TWO3*AuxVar_Grad_i[0][0]);
     residual[nSpecies+1] -= Volume*(yinv*total_viscosity_i*2*(PrimVar_Grad_i[nSpecies+3][1]-v*yinv)
```

Comment:
... for example in line 365 above (`sumJhs_y  += (rho*Ds[iSpecies]*GV[RHOS_INDEX+iSpecies][1] - V_i[RHOS_INDEX+iSpecies]*Vector[1]) * hs[iSpecies];`)  and also 366, where is Vector[1] populated?

Reply:
Its populated in CNEMONumerics.cpp, should I try repopulating it in NEMO_sources?

Reply:
In `GetViscousProjFlux`? How does that value make its way into `CSource_NEMO::ComputeAxisymmetric` ?
Within each application of a public numerics method (like ComputeAxisymmetric or ComputeResidual) you need to be able to trace the result to the inputs, i.e. the things that are set into numerics from the outside before calling the method, or to the constructor, e.g. some constant that is stored in the constructor.
Otherwise you are computing things for one point based on values of another point.

This pattern we have in SU2 in general of "setting" followed by "compute" with very "stateful" objects (i.e. full of member variables used for intermediate results) is not very manageable unfortunately.
We have been getting away with it because most of these class hierarchies are simple, but for richer physics it is going to be headache after headache...

Reply:
Yeah exactly in `GetViscousProjFlux`.  So in cases like this, where there aren't set/compute functions I can just call to update the value, would the best remedy be to explicitly recompute `Vector` in `CSource_NEMO::ComputeAxisymmetric`?

Reply:
Whenever you find yourself needing to do the same action in more than one location, make it a function.
Functions should have clear inputs and outputs.
Programming via side effects, i.e. you call a method of a class and it does some magic to a member variable that you use afterwards is a bad pattern.
The safeties the language offers to keep us from using that pattern are `const` and `static` qualifiers. `const` methods can only read from member variables, `static` can only use other static methods/variables.
"Make everything you can `const`" (every C++ author ever)